provider/source/src/datetime/neo_skeleton.rs

```diff
@@ -89,35 +89,37 @@ impl SourceDataProvider {
 
         fn enforce_consistent_field_lengths_in_pattern<'a>(
             names: &mut FixedCalendarDateTimeNames<()>,
-            runtime_pattern: runtime::Pattern<'a>,
+            mut runtime_pattern: runtime::Pattern<'a>,
         ) -> runtime::Pattern<'a> {
             use icu::datetime::pattern::PatternLoadError;
             use icu::datetime::provider::fields::*;
             use icu::datetime::provider::pattern::PatternItem;
             let date_time_pattern = runtime_pattern.clone().into_date_time_pattern();
-            let Err(e) = names.load_for_pattern(&DebugProvider, &date_time_pattern) else {
-                return runtime_pattern; // nothing to do
-            };
-            let PatternLoadError::ConflictingField {
-                field: requested_field,
-                previous_field,
-            } = e
-            else {
-                panic!("only know how to fix ConflictingField")
-            };
-            let requested_field = Field::from(requested_field);
-            let previous_field = Field::from(previous_field);
-            log::warn!("Rewriting conflicting field: {requested_field:?} --> {previous_field:?}");
-            let mut pattern_items = reference::Pattern::from(&runtime_pattern).into_items();
-            for pattern_item in pattern_items.iter_mut() {
-                let PatternItem::Field(field) = pattern_item else {
-                    continue; // nothing to do
+            while let Err(e) = names.load_for_pattern(&DebugProvider, &date_time_pattern) {
```

Comment:
issue: can we guarantee that this terminates? PErhaps we should document why this terminates OR have some kind of debug tracker that ensures this only runs for X iterations